tests/testthat/test-extract.R

```diff
@@ -0,0 +1,42 @@
+testthat::test_that("`[.` returns empty teal_data for improper names", {
+  data <- teal_data(x = 1, a = 2)
+  testthat::expect_warning(
+    testthat::expect_equal(data["y"], teal_data())
+  )
+})
+
+testthat::test_that("`[.` handles empty names", {
```

Comment:
Handling length of the `names` is handled by `qenv` and this test should be in `teal.code`

Reply:
Alrighty, moved to teal.code